rasa/nlu/training_data/formats/rasa_yaml.py

```diff
@@ -322,3 +350,108 @@ def is_yaml_nlu_file(filename: Text) -> bool:
+        nlu_items.extend(self.process_intents(training_data))
+        nlu_items.extend(self.process_synonyms(training_data))
+        nlu_items.extend(self.process_regexes(training_data))
+        nlu_items.extend(self.process_lookup_tables(training_data))
+
+        result = OrderedDict()
+        result[KEY_TRAINING_DATA_FORMAT_VERSION] = DoubleQuotedScalarString(
+            LATEST_TRAINING_DATA_FORMAT_VERSION
+        )
+        result[KEY_NLU] = nlu_items
+
+        io_utils.write_yaml(result, filename, True)
+
+    @staticmethod
+    def process_intents(training_data: "TrainingData") -> List[OrderedDict]:
+        training_data = TrainingDataWriter.prepare_training_examples(training_data)
+        return RasaYAMLWriter.process_training_examples_by_key(
+            training_data,
+            KEY_INTENT,
+            KEY_INTENT_EXAMPLES,
+            TrainingDataWriter.generate_message,
+        )
+
+    @staticmethod
+    def process_synonyms(training_data: "TrainingData") -> List[OrderedDict]:
+        inverted_synonyms = OrderedDict()
+        for example, synonym in training_data.entity_synonyms.items():
+            if not inverted_synonyms.get(synonym):
+                inverted_synonyms[synonym] = []
+            inverted_synonyms[synonym].append(example)
+
+        return RasaYAMLWriter.process_training_examples_by_key(
+            inverted_synonyms, KEY_SYNONYM, KEY_SYNONYM_EXAMPLES
+        )
+
+    @staticmethod
+    def process_regexes(training_data: "TrainingData") -> List[OrderedDict]:
+        inverted_regexes = OrderedDict()
+        for regex in training_data.regex_features:
+            if not inverted_regexes.get(regex["name"]):
+                inverted_regexes[regex["name"]] = []
+            inverted_regexes[regex["name"]].append(regex["pattern"])
+
+        return RasaYAMLWriter.process_training_examples_by_key(
+            inverted_regexes, KEY_REGEX, KEY_REGEX_EXAMPLES
+        )
+
+    @staticmethod
+    def process_lookup_tables(training_data: "TrainingData") -> List[OrderedDict]:
+        prepared_lookup_tables = OrderedDict()
+        for lookup_table in training_data.lookup_tables:
+            prepared_lookup_tables[lookup_table["name"]] = lookup_table["elements"]
+
+        return RasaYAMLWriter.process_training_examples_by_key(
+            prepared_lookup_tables, KEY_LOOKUP, KEY_LOOKUP_EXAMPLES
+        )
+
+    @staticmethod
+    def process_training_examples_by_key(
+        training_examples: Dict,
+        key_name: Text,
+        key_examples: Text,
+        example_extraction_predicate=lambda x: x,
+    ) -> List[OrderedDict]:
+        from ruamel.yaml.scalarstring import LiteralScalarString
+
+        result = []
+        for entity_key, examples in training_examples.items():
+
+            converted_examples = []
+            converted_examples += [
+                TrainingDataWriter.generate_list_item(
+                    example_extraction_predicate(example).strip("\n\r ")
```

Comment:
what if an example contains new lines?

Reply:
That's a good question! The current implementation of reader/writer is done with an assumption that every line should start with `-` and be a separate training example, e.g. in the reader we're using the `splitlines()` all over.
I guess this has to be noted in the docs and possibly be changed before the release.

Reply:
that sounds good, I think in the future we can make the splitting more advanced, and e.g. only treat lines starting with `-` as a new example. until then we might be able to escape the newlines? (using a literal `\n`)

Reply:
Yes, the `\n` works for now!